lib/hapi-statsd.js

```diff
@@ -19,13 +24,36 @@ module.exports.register = function (server, options, next) {
 		normalizePath = function(path) {
 			path = (path.indexOf('/') === 0) ? path.substr(1) : path;
 			return path.replace(/\//g, settings.pathSeparator);
+		},
+		filterCache = {}, //cache results so that we don't have to compute them every time
+		getFilter = function(statName, route, status) {
+			var filter = filterCache[statName];
+			if(filter) {
+				return filter;
+			}
+
+			var foundFilter = settings.filters.find(function(filter) {
+				// a match on route id
+				if (route.settings && route.settings.id && route.settings.id === filter.id) {
+					return true;
+				}
+
+				return (filter.path || filter.method || filter.status) &&
+					(!filter.path || route.path === filter.path) &&
+					(!filter.method || route.method.toUpperCase() === filter.method.toUpperCase()) &&
+					(!filter.status || status === filter.status);
+			});
+
+			filterCache[statName] = filter = Hoek.applyToDefaults(settings.defaultFilter, foundFilter || {});
+			return filter;
 		};
 
 	server.decorate('server', 'statsd', statsdClient);
 
 	server.ext('onPreResponse', function (request, reply) {
 		var startDate = new Date(request.info.received);
 		var statusCode = (request.response.isBoom) ? request.response.output.statusCode : request.response.statusCode;
+		var method = request.method.toUpperCase();
```

Comment:
I'm keeping with the coding standard of the original module. They don't use const or let.